app/controllers/halls_controller.rb

```diff
@@ -0,0 +1,52 @@
+class HallsController < ApplicationController
+
+    before_action :set_hall, only: [:show, :update, :destroy]
+
+    def index
+        @halls = Hall.all.map do |hall|
+            {
```

Comment:
There's a possibility that your response might change over time and you will be retrieving `Hall's` from some other place in order to unify this you can use something like this [Active Model Serializers](https://wajeeh-ahsan.medium.com/rails-serialization-active-model-serializer-7af16ee9a466)